ui/app/models/ssh/ca-config.js

```diff
@@ -5,16 +5,49 @@
 
 import Model, { attr } from '@ember-data/model';
 import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import { withModelValidations } from 'vault/decorators/model-validations';
 
+const validations = {
+  publicKey: [
+    {
+      validator(model) {
+        const { publicKey, privateKey, generateSigningKey } = model;
+        // if generateSigningKey is false, use the second validation to check if both public and private keys are set
+        return (publicKey && privateKey) || (!publicKey && !privateKey) || !generateSigningKey ? true : false;
```

Comment:
I think I'm a little confused why this validation `!generateSigningKey ? true : false;` is here? it seems like the validation below should be sufficient to handle this. Based on the message, I'd expect this to be: 
```suggestion
        return publicKey || privateKey ? publicKey && privateKey : false ;
```

Reply:
I'll amend the message to clarify. But if generateSigningKey is _true_, then we don't need to check for either public key or private key, they're allowed to be empty. So if I use the suggested logic I would error in that situation, which I don't want to do. 
![image](https://github.com/user-attachments/assets/9fe5b914-4c6f-4084-969c-86bbe135c953)

Reply:
Did a little refactoring of this flow to make it clearer.